server/src/test/java/io/spine/server/projection/ProjectionRepositoryTest.java

```diff
@@ -171,111 +175,121 @@ public void setUp() {
+
+            // Post an event message and grab the ID of the projection, which processed it.
+            checkDispatchesEvent(msg);
+            Set<ProjectId> projectIds = TestProjection.whoProcessed(msg);
+            assertEquals(1, projectIds.size());
+            ProjectId receiverId = projectIds.iterator()
+                                             .next();
+
+            // Check that the projection item has actually been stored and now can be loaded.
+            Iterator<TestProjection> allItems = repository().loadAll();
+            assertTrue(allItems.hasNext());
+            TestProjection storedProjection = allItems.next();
+            assertFalse(allItems.hasNext());
+
+            // Check that the stored instance has the same ID as the instance handling the event.
+            assertEquals(storedProjection.getId(), receiverId);
+        }
 
-    @Test
-    public void dispatch_event_to_archived_projection() {
-        PrjProjectArchived projectArchived = GivenEventMessage.projectArchived();
-        checkDispatchesEvent(projectArchived);
-        ProjectId projectId = projectArchived.getProjectId();
-        TestProjection projection = repository().findOrCreate(projectId);
-        assertTrue(projection.isArchived());
-
-        // Dispatch an event to the archived projection.
-        checkDispatchesEvent(GivenEventMessage.taskAdded());
-        projection = repository().findOrCreate(projectId);
-        List<Task> addedTasks = projection.getState()
-                                          .getTaskList();
-        assertFalse(addedTasks.isEmpty());
-
-        // Check that the projection was not re-created before dispatching.
-        assertTrue(projection.isArchived());
-    }
+        @Test
+        @DisplayName("several events")
+        void severalEvents() {
+            checkDispatchesEvent(GivenEventMessage.projectCreated());
+            checkDispatchesEvent(GivenEventMessage.taskAdded());
+            checkDispatchesEvent(GivenEventMessage.projectStarted());
+        }
 
-    @Test
-    public void dispatch_event_to_deleted_projection() {
-        PrjProjectDeleted projectDeleted = GivenEventMessage.projectDeleted();
-        checkDispatchesEvent(projectDeleted);
-        ProjectId projectId = projectDeleted.getProjectId();
-        TestProjection projection = repository().findOrCreate(projectId);
-        assertTrue(projection.isDeleted());
-
-        // Dispatch an event to the deleted projection.
-        checkDispatchesEvent(GivenEventMessage.taskAdded());
-        projection = repository().findOrCreate(projectId);
-        List<Task> addedTasks = projection.getState()
-                                          .getTaskList();
-        assertTrue(projection.isDeleted());
-
-        // Check that the projection was not re-created before dispatching.
-        assertFalse(addedTasks.isEmpty());
-    }
+        @Test
+        @DisplayName("event to archived projection")
```

Comment:
I think events are NOT dispatched to archived and deleted instances.

Please check. That makes difference to the display name of these tests.